### ISSUE TITLE
the plugin times out with error: "An attempt was made to access a socket in a way forbidden by its access permissions"

### ISSUE THREAD

**Issue by alsadoon101** [tgiesselmann/gimp-stable-boy#26]
Hi,
environment info:
WIN10 
GIMP 2.10.32 (revision 1)
Python 3.10.6
SD WEBUI 1111 is working on a WSL Debian machine with Python 3.9.2
--api is added to COMMANDLINE_ARGS and I am getting the output. "Launching Web UI with arguments: --api"
--api is the only argument added.

SD is working fine as expected.

plugin is installed seemingly fine.

how to reproduce:
simply in plugins text to image, after preparing the prompt and clicking OK
It times out after 2 minutes
error:
<urlopen error [Errno 10013] An attempt was made to access a socket in a way forbidden by its access permissions>
![image](https://user-images.githubusercontent.com/52462725/220974419-fb106f3b-30d5-4738-83de-0cc48d9594c6.png)
![image](https://user-images.githubusercontent.com/52462725/220974531-b15db7f0-013c-41e9-b6c7-37fe1ec7d7c2.png)

also, no output is produced in SD's terminal.

I have firewalls in my machine. I've disabled them just in case but still..
not sure how troubleshoot it.. this is the first time I am using it.

**Comment by alsadoon101** [comment]
the API URL used:
both
http://127.0.0.1:7860/
&
http://localhost:7860

**Comment by thndrbrrr** [comment]
Hi @alsadoon101 – from searching StackOverflow a bit it would seem like the port might already be used by another process? Can't say for sure, I haven't encountered this yet and I don't have Windows.

- Have you tried using another port?
- Does running GIMP as an administrator make a difference?

**Comment by alsadoon101** [comment]
@thndrbrrr yes I've came across such discussions but I don't think that would be my case. otherwise, webui won't open for me.

I've tried both `--port` and running as admin but still. I've tried different combinations with the following parameters but it is still not working:
`export COMMANDLINE_ARGS="--api --api-log --nowebui --port 7870 --listen"
`
interestingly, with `--api-log`, when trying to reach from browser I get 404 error:
```
INFO:     Started server process [3047]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:7870 (Press CTRL+C to quit)
INFO:     127.0.0.1:46074 - "GET / HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:36094 - "GET / HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:36104 - "GET / HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:36104 - "GET / HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:36104 - "GET / HTTP/1.1" 404 Not Found
```

however, from the plugin, nothing is generated in the SD terminal's output.
I am thinking it could be either faulty GIMP installation, or a python misconfiguration. the process that is hanging is pythonw.exe under the path "`path\GIMP 2\bin\pythonw.exe`". I already have in my system python 3.10.6 and its configured under system environment variable's path. So could it be a conflict? which python version is pythonw.exe utilizing?

I think figuring that out will fix the issue.
BTW, running, in cmd, the python file under `path\GIMP 2\bin\python.exe` , shows version 2.7.18
```
path\GIMP 2\bin>python.exe
Python 2.7.18 (default, Oct 27 2021, 07:49:35)  [GCC 11.2.0 64 bit (AMD64)] on win32
```
